### PR TITLE
Minor refactoring in AddExchanges

### PR DESCRIPTION
**Pull request by kokosing**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java [hunk 1]
```diff
@@ -642,10 +642,12 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
 
         private Predicate<TableLayoutResult> layoutHasAllNeededOutputs(TableScanNode node)
         {
-            List<ColumnHandle> columnHandles = node.getOutputSymbols().stream()
+            List<ColumnHandle> nodeColumnHandles = node.getOutputSymbols().stream()
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java]: The variable rename belongs to where the variable was introduced (`Prefer stream usage over Lists.transform in AddExchanges` commit)

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java [hunk 2]
```diff
@@ -642,10 +642,12 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
...
                     .map(node.getAssignments()::get)
                     .collect(toImmutableList());
-            return layout -> !layout.getLayout().getColumns().isPresent() || layout.getLayout().getColumns().get().containsAll(columnHandles);
+            return layout -> layout.getLayout().getColumns()
+                    .map(columnHandles -> columnHandles.containsAll(nodeColumnHandles))
+                    .orElse(true);
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java]: I find previous form more readable.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java [hunk 3]
```diff
@@ -640,12 +639,12 @@ private PlanWithProperties planTableScan(TableScanNode node, Expression predicat
             return pickPlan(possiblePlans, context);
         }
 
-        private Predicate<TableLayoutResult> layoutHasAllNeededOutputs(TableScanNode node)
+        private boolean layoutHasAllNeededOutputs(TableLayoutResult layout, TableScanNode node)
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java]: You can now rename to `hasAllNeededOutputs` or `hasAllOutputsOfNode`

**Comment by kokosing** [comment]
Comments applied

**Pull request merged**